graphs/uniform_search_cost.py

```diff
@@ -0,0 +1,260 @@
+# diagonal clockwise
+dxy1 = [
+    [1, 1],
+    [1, 0],
+    [1, -1],
+    [0, -1],
+    [-1, -1],
+    [-1, 0],
+    [-1, 1],
+    [0, 1],
+]
+# diagonal anti-clockwise
+dxy2 = [
+    [-1, -1],
+    [-1, 0],
+    [0, -1],
+    [0, 1],
+    [1, 1],
+    [1, 0],
+    [1, -1],
+    [-1, 1],
+]
+
+# start point and end point on same row and column right side
+dxy3 = [
+    [0, -1],
+    [-1, -1],
+    [-1, 0],
+    [0, 1],
+    [1, 1],
+    [1, 0],
+    [1, -1],
+    [-1, 1],
+]
+# start point and end point on same row and column left side
+dxy4 = [
+    [0, 1],
+    [1, 1],
+    [1, 0],
+    [1, -1],
+    [0, -1],
+    [-1, -1],
+    [-1, 0],
+    [-1, 1],
+]
+# start point and end point on same column and row down side
+dxy5 = [
+    [1, 0],
+    [0, 1],
+    [1, 1],
+    [1, -1],
+    [0, -1],
+    [-1, -1],
+    [-1, 0],
+    [-1, 1],
+]
+# start point and end point on same column and row up side
+dxy6 = [
+    [0, -1],
+    [0, 1],
+    [1, 1],
+    [1, 0],
+    [1, -1],
+    [-1, -1],
+    [-1, 0],
+    [-1, 1],
+]
+
+
+class UniformCostSearch:
+    def __init__(self, grid: list[list[int]]) -> None:
+        self.m = len(grid[0])
+        self.n = len(grid)
+
+    def get_shortest_path(
+        self,
+        start: list[int],
+        end: list[int],
+        dist: list[list[float]],
+        dxy: list[list[int]],
+    ) -> list[list[float]] | list[list[int]]:
+        """
+        Return 2D list where optimal path is stored.
+        # >>> start = [0,0]
+        # >>> end = [1, 2]
+        # >>> dist = [[1,0],[0,2]]
+        # >>> dxy = [(1, 1),(1, 0),(1, -1),(0, -1),(-1, -1),(-1, 0),(-1, 1),(0, 1)]
+        # >>> get_shortest_path(start,end,dist,dxy)
+        # [[0,0],[1,1]]
+        """
```

Comment:
why these doctests are commented?

Reply:
Actually, i was facing problem to run algorithm in this repository action CI. I will uncomment and check it.